src/game/game/game.c

```diff
@@ -207,12 +216,12 @@ int32_t Game_Stop(void)
     if (g_CurrentLevel == g_GameFlow.last_level_num) {
         g_Config.profile.new_game_plus_unlock = true;
         Config_Write();
-    } else {
-        Savegame_CarryCurrentInfoToNextLevel(
-            g_CurrentLevel, g_CurrentLevel + 1);
-        Savegame_ApplyLogicToCurrentInfo(g_CurrentLevel + 1);
     }
 
+    Savegame_CarryCurrentInfoToNextLevel(
```

Comment:
I understand that this `else` was removed to have the game carry over Lara's inventory and stats to the bonus levels as well, which usually happen to be placed after the normal levels.

What will happen, however, if the game has no bonus levels, and we've just ended the last level? I think it will create some memory corruption trying to carry the stats over to a non-existing item. Neither `Savegame_ApplyLogicToCurrentInfo` nor `Savegame_CarryCurrentInfoToNextLevel` check if the target level number is valid, so we either should make them more resilient, or convert this `else` here to `if (g_CurrentLevel + 1 < g_GameFlow.level_count)`.

Reply:
Pushed your second suggestion.